### DISCUSSION TITLE
functions output missing `next-server` when using `vercel build` and `vercel deploy --prebuilt` (locally or from github actions)

### DISCUSSION THREAD

**Discussion by IvanCoronado** [vercel/vercel#8294] [Help]
I have a monorepo with turborepo. I am trying to build and deploy using the `vercel cli` from a Github action.

First run `vercel pull {...} && vercel build {...}`
After run `vercel deploy --prebuilt {...}`

All is working without errors but all the code in `getStaticProps` is not working. `Cannot find module 'next/dist/server/next-server.js'`

<img width="1271" alt="image" src="https://user-images.githubusercontent.com/13351933/182445914-e736d599-a728-4517-88ca-e4badf6d061f.png">

Looks like functions are missing `node_modules` in the output code generated.
<img width="1233" alt="image" src="https://user-images.githubusercontent.com/13351933/182445844-3196455e-bb55-4d15-b265-be0430ca5dc2.png">

**Comment by amyegan** [comment]
Hi @IvanCoronado. It looks like this could be related to a build cache issue that some other people have reported in https://github.com/vercel/turborepo/discussions/1065. What happens if you use `--force` to bypass the cache as suggested in https://github.com/vercel/turborepo/discussions/1065#discussioncomment-2969678?

Please share any more details that you can (i.e. your repo, a minimal reproduction, or your config files). Any additional information you provide can be very helpful for us to figure out what's happening.

**Reply by IvanCoronado** [reply]
Hi @amyegan!

I was already using `cache: false` and still having the problem.

I have recreated my workflow using turborepo examples and have the same problem.

Repo -> https://github.com/IvanCoronado/turborepo-test-build-npm 
Vercel project config -> 
<img width="939" alt="image" src="https://user-images.githubusercontent.com/13351933/182554684-5cba8652-d0ff-41d0-a74d-e46ae4be106f.png">
Run `npx vercel pull` in `apps/web` and connect the repo with the Vercel project
Run `npm run build` and `npm run deploy` in `./`
<img width="895" alt="image" src="https://user-images.githubusercontent.com/13351933/182556079-0bb7311b-7d1a-4f51-afd6-5e83dd9169b3.png">
Open the deployed URL and navigate to `/test/a` and you get the next error
<img width="1096" alt="image" src="https://user-images.githubusercontent.com/13351933/182556396-f55c0553-9ba3-492e-b3f4-fe96c9052acd.png">
<img width="1260" alt="image" src="https://user-images.githubusercontent.com/13351933/182556769-8aa67108-e420-4bf7-945f-97b30bf78d5d.png">

**Reply by IvanCoronado** [reply]
I have found a workaround calling the build/deploy from the root folder.

Demo repo here -> https://github.com/IvanCoronado/turborepo-test-build-npm/tree/build-from-root
Vercel project config -> 
<img width="920" alt="image" src="https://user-images.githubusercontent.com/13351933/182587441-6392241c-8b67-40e0-92a3-c0944e9673be.png">
Run `npx vercel pull` in `./` and connect the repo with the Vercel project
Run `npm run vercel-build` and `npm run vercel-deploy` in `./`
And it's working 🎉

The problem with this solution is that I can only deploy 1 repo to vercel for monorepo.

**Reply by amyegan** [reply]
Thanks for sharing these examples @IvanCoronado. I asked the Turborepo engineers and they suggested making a change to your Root Directory. Does it work if you change the Root Directory from `./` to `apps/web` for your [turborepo-test-build-npm](https://github.com/IvanCoronado/turborepo-test-build-npm) example?

**Reply by IvanCoronado** [reply]
@amyegan 

using this repo -> https://github.com/IvanCoronado/turborepo-test-build-npm
and this config -> 
<img width="943" alt="image" src="https://user-images.githubusercontent.com/13351933/182648655-382a0ab2-6235-4c10-94b3-3a39ddf4fd21.png">
Run `npx vercel pull` in `apps/web` and connect the repo with the Vercel project
Run `npm run build` from root throw this error
<img width="1404" alt="image" src="https://user-images.githubusercontent.com/13351933/182649329-e38d6624-d162-4881-b54c-74d97a13adcf.png">

I think that `vercel build` is getting confused on some point by the Root Directory and the mono repo when I run it on local.

**Reply by amyegan** [reply]
The Engineering team is investigating, but we have one more thing to try while we wait. This could be caused by a build order problem. If you add environment variable `ENABLE_VC_DEPLOY=1`, that should get your Project using a slightly different build setup. This has solved similar problems recently.

**Reply by IvanCoronado** [reply]
thanks, @amyegan and Vercel team.

Using this https://github.com/vercel/vercel/discussions/8294#discussioncomment-3314144

And running `ENABLE_VC_DEPLOY=1 npm run deploy` but still not working.

Same error -> https://turborepo-test-build-nl8nch0cq-dividivan.vercel.app/test/a

**Reply by nathanhammond** [reply]
The related issue we're referring to is #8063.

We've [not been able to reproduce your issue](https://turborepo-test-build-npm-p7gwnq65r-turbo-test-vtest314.vercel.app/) with the flag on and these settings:

<img width="958" alt="CleanShot 2022-08-03 at 12 32 22@2x" src="https://user-images.githubusercontent.com/20542/182678714-cdf50739-fcb6-4d9e-8a6b-9ab929035047.png">

Also, I'd generally recommend configuring `ENABLE_VC_DEPLOY` in the Vercel environment variable interface.

**Reply by nathanhammond** [reply]
It's also possible that the `ENABLE_VC_DEPLOY` flag is not enabled for your account. If that's the case we can get you opted into the flag to unblock you. Can you pass me your team name?

**Reply by IvanCoronado** [reply]
thanks you @nathanhammond, still getting same error with the `ENABLE_VC_DEPLOY`

My team's name is `wuolah`

**Reply by nathanhammond** [reply]
Okay, I think your account is dividivan, we're opting you into that flag, give us a minute.

**Reply by IvanCoronado** [reply]
Yep, but I need it for my company team. I used `dividivan` only for creating the demo.

**Reply by nathanhammond** [reply]
What's the company account?

**Reply by nathanhammond** [reply]
Saw the edit!

**Reply by IvanCoronado** [reply]
@nathanhammond not sure if we have the flag already activated but have you deployed [this](https://turborepo-test-build-npm-p7gwnq65r-turbo-test-vtest314.vercel.app/) from your local with the `--prebuilt` in the  `vercel deploy`?  

I have connected the repo to my company Vercel team with this config and [build it properly](https://turborepo-test-build-npm-six.vercel.app/) from the Vercel context.
<img width="973" alt="image" src="https://user-images.githubusercontent.com/13351933/182692774-0e32ee76-df21-45ed-aec9-376e53de09d8.png">

But still getting this error when trying to build from my local 😕 
<img width="1421" alt="image" src="https://user-images.githubusercontent.com/13351933/182693401-11a52b22-b83c-467f-ac14-515a6bf85692.png">

**Reply by amyegan** [reply]
Thanks for your patience with all of this. I have some new info to share.
1. I gave you the wrong environment variable name, and I am terribly sorry about that. The flag is enabled for your team/account now. If you add `ENABLE_VC_BUILD` with value `1` in the Vercel environment variable interface it will sort out potential build order issues.
2. You actually _should_ be running build from the root folder like you did in your workaround example. Then you can use the other configuration options to signify which directory your project should deploy, etc.
